pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -1352,6 +1393,238 @@ func (vs *VSphere) NodeExists(c *govmomi.Client, nodeName k8stypes.NodeName) (bo
 	return false, nil
 }
 
+func (vs *VSphere) createDummyVMWithDiskAttached(ctx context.Context, datacenter *object.Datacenter, datastore *object.Datastore, volumeOptions *VolumeOptions) (string, error) {
+	var diskFormat string
+
+	// Default diskformat as 'thin'
+	if volumeOptions.DiskFormat == "" {
+		volumeOptions.DiskFormat = ThinDiskType
+	}
+
+	if _, ok := diskFormatValidType[volumeOptions.DiskFormat]; !ok {
+		return "", fmt.Errorf("Cannot create disk. Error diskformat %+q."+
+			" Valid options are %s.", volumeOptions.DiskFormat, DiskformatValidOptions)
+	}
+
+	diskFormat = diskFormatValidType[volumeOptions.DiskFormat]
+
+	// Generate a UUID
+	uuidVal := string(uuidgenerator.NewUUID())
+	dummyVMName := "Kubernetes-worker-node" + uuidVal
+
+	virtualMachineConfigSpec := types.VirtualMachineConfigSpec{
+		Name: dummyVMName,
+		Files: &types.VirtualMachineFileInfo{
+			VmPathName: "[" + datastore.Name() + "]",
+		},
+		NumCPUs:  1,
+		MemoryMB: 4,
+	}
+
+	scsiDeviceConfigSpec := &types.VirtualDeviceConfigSpec{
+		Operation: types.VirtualDeviceConfigSpecOperationAdd,
+		Device: &types.VirtualLsiLogicController{
+			types.VirtualSCSIController{
+				SharedBus: types.VirtualSCSISharingNoSharing,
+				VirtualController: types.VirtualController{
+					BusNumber: 0,
+					VirtualDevice: types.VirtualDevice{
+						Key: 1000,
+					},
+				},
+			},
+		},
+	}
+
+	diskConfigSpec := &types.VirtualDeviceConfigSpec{
+		Operation:     types.VirtualDeviceConfigSpecOperationAdd,
+		FileOperation: types.VirtualDeviceConfigSpecFileOperationCreate,
+	}
+
+	virtualDiskSpec := &types.VirtualDisk{
+		CapacityInKB: int64(volumeOptions.CapacityKB),
+	}
+	virtualDeviceSpec := types.VirtualDevice{
+		Key:           0,
+		ControllerKey: 1000,
+		UnitNumber:    new(int32), // zero default value
+	}
+
+	backingObjectSpec := &types.VirtualDiskFlatVer2BackingInfo{
+		DiskMode: string(types.VirtualDiskModePersistent),
+		VirtualDeviceFileBackingInfo: types.VirtualDeviceFileBackingInfo{
+			FileName: "[" + datastore.Name() + "] " + volumeOptions.Name + ".vmdk",
```

Comment:
As we discussed, if you can't specify Datastore\kubevols\volumeOptions.Name + ".vmdk"   then let's see if we can separate CreateVM and AddDisk operation.